homeassistant/components/synology_srm/device_tracker.py

```diff
@@ -73,6 +73,14 @@ def scan_devices(self):
 
         return [device["mac"] for device in self.last_results]
 
+    def get_extra_attributes(self, device) -> dict:
+        """Get the extra attributes of a device."""
+        filter_result = [
+            result for result in self.last_results if result["mac"] == device
+        ]
+        if filter_result:
+            return filter_result[0]
```

Comment:
I would suggest to use the following pattern for this method:
https://github.com/home-assistant/core/blob/16336bf9021ac1687180c586319c9e908eee6ead/homeassistant/components/ee_brightbox/device_tracker.py#L94-L102

Making sure that an empty dict is returned when nothing is found and if something is found returning a dict with the proper information. Is this taken into account here as well?

Reply:
No it wasn't ... was an over site.